src/test/unit/controllers/FlatFileController.test.ts

```diff
@@ -0,0 +1,90 @@
+import sinon from 'sinon';
+import {Response} from 'express';
+import {PublicationService} from '../../../main/service/publicationService';
+import FlatFileController from '../../../main/controllers/FlatFileController';
+import {PipRequest} from '../../../main/models/request/PipRequest';
+
+const metaStub = sinon.stub(PublicationService.prototype, 'getIndivPubMetadata');
+const fileStub = sinon.stub(PublicationService.prototype, 'getIndivPubFile');
+const mockFile = new Blob(['testFile']);
+
+const response = {
+  send: () => {
+    return '';
+  },
+  set: () => {
+    return '';
+  }} as unknown as Response;
+
+const errResponse = {
+  render: () => {
+    return '';
+  },
+  set: () => {
+    return '';
+  }} as unknown as Response;
+
+describe('Flat File Controller', () => {
+  const flatFileController = new FlatFileController();
+
+  it('should return a pdf when appropriate', () => {
+    metaStub.withArgs(0).resolves({'isFlatFile':'true', 'sourceArtefactId':'doc.pdf'});
+    fileStub.withArgs(0).resolves(mockFile);
+    const request = {
+      query: {'artefactId': 0}, i18n: {
+        getDataByLanguage: '',
+      },
+    } as unknown as PipRequest;
+    request.i18n.getDataByLanguage = jest.fn().mockReturnValue('mockI18n');
```

Comment:
this is unnecessary, use the same way we have in all our other controller tests mocking this part
```
const request = mockRequest(i18n);
```

Reply:
applies to all